### DISCUSSION TITLE
Governance Stage 0: MCR, Staking, and Token Smart Contracts

### DISCUSSION THREAD

**Discussion by l-monninger** [movementlabsxyz/movement#433] [Ideas]
# Summary
For simplicity, the current proposal for Governance Stage 0 is twofold is to use a number of wallets $n$ and request $n$ of $n$, i.e., unanimous, voting to for any proposal. Proposals would be managed through an OpenZeppelin [TimelockController](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/TimelockController.sol) and the multisig wallet can take form similar to [`Safe`](https://github.com/safe-global/safe-smart-account/blob/main/contracts/Safe.sol).

**Comment by l-monninger** [comment]
#435 and MD-2 cover an approaches that rely on wallet patterns here.

**Comment by Primata** [comment]
https://github.com/movementlabsxyz/movement/blob/5c20d360a484bced6fa0499b7b998fd055d99bc7/protocol-units/settlement/mcr/contracts/script/DeployMCRStaking.s.sol#L37

Proposed design of 5 proposers (engineers) and executor being the multisig address.

**Reply by l-monninger** [reply]
Yeah, I believe this works. Let me sketch something for you.

**Comment by l-monninger** [comment]
FYI, we can go back to https://github.com/gnosis/MultiSigWallet/blob/master/contracts/MultiSigWallet.sol

We don't have to use SafeSmart accounts if stripping out assumptions from Gnosis Safe is too difficult.
